connectors/source.py

```diff
@@ -107,6 +117,111 @@ def is_empty(self):
     def to_dict(self):
         return dict(self._raw_config)
 
+    def is_valid(self):
+        """Validates every Field against its `validations`.
+
+        Raises ConfigurableFieldValueError if any validation errors are found,
+        otherwise returns True.
+        """
+        validation_errors = []
+
+        for _, field in self._config.items():
+            if not self.dependencies_satisfied(field):
+                # we don't validate a field if its dependencies are not met
+                next
+
+            validation_errors = validation_errors + self.validate_field(field)
+
+        if len(validation_errors) > 0:
+            raise ConfigurableFieldValueError(
+                f"Field validations: a{'; '.join(validation_errors)}"
+            )
+
+        return True
+
+    def dependencies_satisfied(self, field):
+        """Used to check if a Field has its dependencies validated.
+
+        Returns True if all dependencies are satisfied, or no dependencies exist.
+        Retruns False if one or more dependencies are not satisfied.
+        """
+        if len(field.depends_on) <= 0:
+            return True
+
+        satisfied = True
+
+        for dependency in field.depends_on:
+            if dependency["field"] not in self._config:
+                # cannot check dependency if field does not exist
+                logger.warn(
```

Comment:
I think that could be an `error`, right? As this only occurs, when someone provides a wrong implementation.

Reply:
True, if we're throwing errors for everything else we should be consistently strict here.

Reply:
I've thought to change the log-level to `error`, but it's probably also valid to just raise an error.

Reply:
Added, also I added more tests to check that this is handled appropriately.